aptos-move/framework/aptos-framework/sources/PropertyList.move

```diff
@@ -0,0 +1,175 @@
+/// This Data Structure is designed to hold properties.
+/// It only support
+///     1. adding new property.
+///     2. update the value of existing property
+///     3. get the value of a property
+module AptosFramework::PropertyList {
+    use Std::Vector;
+    use Std::ASCII::String;
+    use AptosFramework::SimpleMap::{Self, SimpleMap};
+
+    const MAX_PROPERTY_MAP_SIZE: u64 = 1000;
+    const EKEY_AREADY_EXIST_IN_PROPERTY_MAP: u64 = 1;
+    const EPROPERTY_NUMBER_EXCEED_LIMIT: u64 = 2;
+    const EPROPERTY_NOT_EXIST: u64 = 3;
+    const EKEY_COUNT_NOT_MATCH_VALUE_COUNT: u64 = 4;
+    const EKEY_COUNT_NOT_MATCH_TYPE_COUNT: u64 = 5;
+
+    struct PropertyList has store {
+       map: SimpleMap<String, PropertyValue>
+    }
+
+    struct PropertyValue has store, copy, drop {
+        value: vector<u8>,
+        type: String,
+    }
```

Comment:
I guess this is the thing we really need to nail down, do we really want to specify a `type` here. It feels a little worrisome to me. Also as a string, there is no canonical representation on-chain. Personally I'd like to just remove it, but assuming we don't... let's grab the opinions of

@CapCap 
@areshand 
@hariria 
@movekevin 

we go with the majority.